websubmit/Bibtex.py

```diff
@@ -113,10 +113,12 @@ def process_references(references, output_format):
             ref = re.sub(r'\.', ',', ref)
         elif re.search(r'\w\-\w', ref):
             index = 'r'
+        elif re.search(r'\d{4}[\w\.&]{15}', ref):
+            index = 'ads'
         if index:
             # hack to match more records
             recid_list = ''
-            if index == 'texkey':
+            if index == 'texkey' or index == 'ads':
```

Comment:
is there ever an ADS bibcode in `035__z` ? also, why is `035__z` searched before `035__a`? isn't subfiled `a` the current report number?
is it even necessary to search in two steps? why not search `reportnumber:...`  that index contains `037__a/z` 
would there be a different match for searching `035__a` before `035__z` ?

Reply:
correction:
reportnumber index looks in `037`
this needs to look in `035`

so `perform_request_search(p='035:"<searchstring>"')`
or if it needs to be more specific `a/z` subfields only
`perform_request_search(p='035__a:"<searchstring>" OR 035__z:"<searchstring>"')`
for a single lookup instead of 2 separate lookups
also, test if result has exactly 1 recid. if there are multiple matches metadata is likely wrong